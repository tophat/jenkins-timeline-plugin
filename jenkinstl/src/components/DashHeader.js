import React from 'react'

import { Row, Label, Container, TopBar, Title, LogoBox, BackButton } from './DashHeader.style'
import { statusMap } from '../constants'
import Logo from '../assets/logo.png'

export default class DashHeader extends React.PureComponent {
    formatTime = timestamp => {
        const seconds = parseInt(timestamp / 1000) % 60
        const minutes = parseInt(timestamp / (1000 * 60)) % 60

        const parts = []

        if (minutes >= 1) {
            const plural = minutes > 1 ? `s` : ``
            parts.push(`${minutes} minute${plural}`)
        }

        if (seconds >= 1) {
            const plural = seconds > 1 ? `s` : ``
            parts.push(`${seconds} second${plural}`)
        }

        return parts.join(', ')
    }

    render() {
        const formattedTime = this.formatTime(this.props.duration)
        const formattedStatus = statusMap[this.props.buildStatus]
        const buildStatus = `Status: ${formattedStatus}`
        const startTime = `Started on ${this.props.startTime}`
        const runningTimePrefix = this.props.buildStatus === 'IN_PROGRESS' ? `Running` : `Ran`
        const runningTime = `${runningTimePrefix} for ${formattedTime}`

        const backLink = <BackButton href={this.props.buildUrl}>Back to Jenkins</BackButton>

        return (
            <Container
                status={this.props.buildStatus}
            >
                <TopBar>
                    <LogoBox>
                        <img
                            alt="Build timeline"
                            src={Logo}
                        />
                        <Title>Build timeline</Title>
                    </LogoBox>
                    <Label>{backLink}</Label>
                </TopBar>
                <Row>
                    <Label>{buildStatus}</Label>
                    <Label>{startTime}</Label>
                    <Label>{runningTime}</Label>
                </Row>
                <Row>
                    
                </Row>
            </Container>
        )
    }
}