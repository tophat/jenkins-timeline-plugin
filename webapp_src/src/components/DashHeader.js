import React from 'react'
import PropTypes from 'prop-types'
import moment from 'moment'

import Logo from '../assets/logo.png'
import { buildStatuses, statusMap } from '../constants'

import {
    BackButton,
    Container,
    Label,
    LogoBox,
    Row,
    Title,
    TopBar,
} from './DashHeader.style'

export default class DashHeader extends React.PureComponent {
    static propTypes = {
        buildStatus: PropTypes.string,
        buildUrl: PropTypes.string.isRequired,
        buildName: PropTypes.string,
        duration: PropTypes.number,
        endTime: PropTypes.instanceOf(moment),
        longestStage: PropTypes.shape({
            title: PropTypes.string,
            duration: PropTypes.number,
        }),
        startTime: PropTypes.instanceOf(moment),
    }

    static defaultProps = {
        buildName: '',
        buildStatus: buildStatuses.NOT_AVAILABLE,
        duration: 0,
        endTime: null,
        longestStage: null,
        startTime: null,
    }

    formatTime = timestamp => {
        const seconds = parseInt(timestamp / 1000) % 60
        const minutes = parseInt(timestamp / (1000 * 60)) % 60
        const hours = parseInt(timestamp / (1000 * 3600))

        const parts = []

        if (hours >= 1) {
            const plural = hours > 1 ? 's' : ''
            parts.push(`${hours} hour${plural}`)
        }

        if (minutes >= 1) {
            const plural = minutes > 1 ? 's' : ''
            parts.push(`${minutes} minute${plural}`)
        }

        if (seconds >= 1) {
            const plural = seconds > 1 ? 's' : ''
            parts.push(`${seconds} second${plural}`)
        }

        return parts.join(', ')
    }

    onBackButtonClick = () => {
        window.location.assign(this.props.buildUrl)
    }

    getTopBar = () => (
        <TopBar>
            <LogoBox>
                <img alt="Build timeline" src={Logo} />
                <Title>{`Build timeline > ${this.props.buildName}`}</Title>
            </LogoBox>
            <BackButton
                onClick={this.onBackButtonClick}
                href={this.props.buildUrl}
            >
                Back to Jenkins
            </BackButton>
        </TopBar>
    )

    getLongestStageLabel = () => {
        if (!this.props.longestStage) return null
        const longestStage = `Longest stage: ${
            this.props.longestStage.title
        } (${this.formatTime(this.props.longestStage.duration)})`

        return <Label clickId="timeline longest stage">{longestStage}</Label>
    }

    getStartTimeLabel = () => {
        if (!this.props.startTime) return null
        const formattedStartTime = this.props.startTime.format(
            'MMMM Do YYYY, hh:mm:ss a',
        )
        const startTime = `Started on ${formattedStartTime}`
        return <Label clickId="timeline start time">{startTime}</Label>
    }

    getEndTimeLabel = () => {
        if (!this.props.endTime) return null
        const formattedEndTime = this.props.endTime
            ? this.props.endTime.format('MMMM Do YYYY, hh:mm:ss a')
            : null
        const endedOn = this.props.endTime
            ? `Ended on ${formattedEndTime}`
            : null
        return <Label clickId="timeline end time">{endedOn}</Label>
    }

    getRunningTimeLabel = () => {
        if (!this.props.duration) return null
        const runningTimePrefix =
            this.props.buildStatus === buildStatuses.IN_PROGRESS
                ? 'Running'
                : 'Ran'
        const runningTime = `${runningTimePrefix} for ${this.formatTime(
            this.props.duration,
        )}`
        return <Label clickId="timeline running time">{runningTime}</Label>
    }

    getBuildStatusLabel = () => {
        const buildStatus = `Status: ${statusMap[this.props.buildStatus]}`
        return <Label clickId="timeline build status">{buildStatus}</Label>
    }

    getDetailsBar = () => {
        return (
            <React.Fragment>
                <Row>
                    {this.getBuildStatusLabel()}
                    {this.getStartTimeLabel()}
                    {this.getRunningTimeLabel()}
                </Row>
                <Row>
                    {this.getLongestStageLabel()}
                    {this.getEndTimeLabel()}
                    <Label />
                </Row>
            </React.Fragment>
        )
    }

    render() {
        return (
            <Container status={this.props.buildStatus}>
                {this.getTopBar()}
                {this.getDetailsBar()}
            </Container>
        )
    }
}
