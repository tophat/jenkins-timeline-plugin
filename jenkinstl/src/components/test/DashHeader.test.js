import { shallow, mount } from 'enzyme'
import React from 'react'

import DashHeader from '../DashHeader'

describe('DashHeader', () => {
    it('renders the component', () => {
        const header = shallow(<DashHeader/>)
        expect(header).toMatchSnapshot()
    })
})