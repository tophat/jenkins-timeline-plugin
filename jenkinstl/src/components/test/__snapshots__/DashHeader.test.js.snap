// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashHeader renders the component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DashHeader />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <ForwardRef>
          <ForwardRef>
            <img
              alt="Build timeline"
              src="logo.png"
            />
            <ForwardRef>
              Build timeline
            </ForwardRef>
          </ForwardRef>
          <ForwardRef>
            <ForwardRef>
              Back to Jenkins
            </ForwardRef>
          </ForwardRef>
        </ForwardRef>,
        <ForwardRef>
          <ForwardRef>
            Status: undefined
          </ForwardRef>
          <ForwardRef>
            Started on undefined
          </ForwardRef>
          <ForwardRef>
            Ran for 
          </ForwardRef>
        </ForwardRef>,
        <ForwardRef>
          <ForwardRef>
            Status: undefined
          </ForwardRef>
          <ForwardRef>
            Started on undefined
          </ForwardRef>
          <ForwardRef>
            Ran for 
          </ForwardRef>
        </ForwardRef>,
      ],
      "status": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <ForwardRef>
              <img
                alt="Build timeline"
                src="logo.png"
              />
              <ForwardRef>
                Build timeline
              </ForwardRef>
            </ForwardRef>,
            <ForwardRef>
              <ForwardRef>
                Back to Jenkins
              </ForwardRef>
            </ForwardRef>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <img
                  alt="Build timeline"
                  src="logo.png"
                />,
                <ForwardRef>
                  Build timeline
                </ForwardRef>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "Build timeline",
                  "src": "logo.png",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": "Build timeline",
                },
                "ref": null,
                "rendered": "Build timeline",
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bZQynM",
                    "isStatic": true,
                    "rules": Array [
                      "
    margin-left: 10px;
    width: 100%;
",
                    ],
                  },
                  "displayName": "styled.span",
                  "render": [Function],
                  "styledComponentId": "sc-bZQynM",
                  "target": "span",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                },
              },
            ],
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-gzVnrw",
                "isStatic": true,
                "rules": Array [
                  "
    display: flex;
    align-items: center;
",
                ],
              },
              "displayName": "styled.div",
              "render": [Function],
              "styledComponentId": "sc-gzVnrw",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <ForwardRef>
                Back to Jenkins
              </ForwardRef>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Back to Jenkins",
                "href": undefined,
              },
              "ref": null,
              "rendered": "Back to Jenkins",
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-htoDjs",
                  "isStatic": true,
                  "rules": Array [
                    "
    padding: 10px;
    border: 3px solid rgba(0,0,0,0.3);
    background-color: transparent;
    cursor: pointer;
    transition: 0.5s;
    color: ",
                    "#fff",
                    ";

    &:hover {
        background-color: rgba(0,0,0,0.3);
        border-color: transparent;
    }
",
                  ],
                },
                "displayName": "styled.button",
                "render": [Function],
                "styledComponentId": "sc-htoDjs",
                "target": "button",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              },
            },
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-bxivhb",
                "isStatic": true,
                "rules": Array [
                  "
",
                ],
              },
              "displayName": "styled.span",
              "render": [Function],
              "styledComponentId": "sc-bxivhb",
              "target": "span",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-EHOje",
            "isStatic": true,
            "rules": Array [
              "
    box-shadow: inset 9999px 0 0 0 rgba(0, 0, 0, 0.2);
    padding: 10px;
    display: flex;
    align-items: center;
    justify-content: space-between;
",
            ],
          },
          "displayName": "styled.div",
          "render": [Function],
          "styledComponentId": "sc-EHOje",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <ForwardRef>
              Status: undefined
            </ForwardRef>,
            <ForwardRef>
              Started on undefined
            </ForwardRef>,
            <ForwardRef>
              Ran for 
            </ForwardRef>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Status: undefined",
            },
            "ref": null,
            "rendered": "Status: undefined",
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-bxivhb",
                "isStatic": true,
                "rules": Array [
                  "
",
                ],
              },
              "displayName": "styled.span",
              "render": [Function],
              "styledComponentId": "sc-bxivhb",
              "target": "span",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Started on undefined",
            },
            "ref": null,
            "rendered": "Started on undefined",
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-bxivhb",
                "isStatic": true,
                "rules": Array [
                  "
",
                ],
              },
              "displayName": "styled.span",
              "render": [Function],
              "styledComponentId": "sc-bxivhb",
              "target": "span",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Ran for ",
            },
            "ref": null,
            "rendered": "Ran for ",
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-bxivhb",
                "isStatic": true,
                "rules": Array [
                  "
",
                ],
              },
              "displayName": "styled.span",
              "render": [Function],
              "styledComponentId": "sc-bxivhb",
              "target": "span",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": true,
            "rules": Array [
              "
    display: flex;
    justify-content: space-between;
    padding: 10px;
",
            ],
          },
          "displayName": "styled.div",
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <ForwardRef>
              Status: undefined
            </ForwardRef>,
            <ForwardRef>
              Started on undefined
            </ForwardRef>,
            <ForwardRef>
              Ran for 
            </ForwardRef>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Status: undefined",
            },
            "ref": null,
            "rendered": "Status: undefined",
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-bxivhb",
                "isStatic": true,
                "rules": Array [
                  "
",
                ],
              },
              "displayName": "styled.span",
              "render": [Function],
              "styledComponentId": "sc-bxivhb",
              "target": "span",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Started on undefined",
            },
            "ref": null,
            "rendered": "Started on undefined",
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-bxivhb",
                "isStatic": true,
                "rules": Array [
                  "
",
                ],
              },
              "displayName": "styled.span",
              "render": [Function],
              "styledComponentId": "sc-bxivhb",
              "target": "span",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Ran for ",
            },
            "ref": null,
            "rendered": "Ran for ",
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-bxivhb",
                "isStatic": true,
                "rules": Array [
                  "
",
                ],
              },
              "displayName": "styled.span",
              "render": [Function],
              "styledComponentId": "sc-bxivhb",
              "target": "span",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": true,
            "rules": Array [
              "
    display: flex;
    justify-content: space-between;
    padding: 10px;
",
            ],
          },
          "displayName": "styled.div",
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
    background-color: ",
          [Function],
          ";
    color: ",
          "#fff",
          ";

    font-family: 'Source Sans Pro', sans-serif;
    display: flex;
    flex-direction: column;
",
        ],
      },
      "displayName": "styled.header",
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "header",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ForwardRef>
            <ForwardRef>
              <img
                alt="Build timeline"
                src="logo.png"
              />
              <ForwardRef>
                Build timeline
              </ForwardRef>
            </ForwardRef>
            <ForwardRef>
              <ForwardRef>
                Back to Jenkins
              </ForwardRef>
            </ForwardRef>
          </ForwardRef>,
          <ForwardRef>
            <ForwardRef>
              Status: undefined
            </ForwardRef>
            <ForwardRef>
              Started on undefined
            </ForwardRef>
            <ForwardRef>
              Ran for 
            </ForwardRef>
          </ForwardRef>,
          <ForwardRef>
            <ForwardRef>
              Status: undefined
            </ForwardRef>
            <ForwardRef>
              Started on undefined
            </ForwardRef>
            <ForwardRef>
              Ran for 
            </ForwardRef>
          </ForwardRef>,
        ],
        "status": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <ForwardRef>
                <img
                  alt="Build timeline"
                  src="logo.png"
                />
                <ForwardRef>
                  Build timeline
                </ForwardRef>
              </ForwardRef>,
              <ForwardRef>
                <ForwardRef>
                  Back to Jenkins
                </ForwardRef>
              </ForwardRef>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <img
                    alt="Build timeline"
                    src="logo.png"
                  />,
                  <ForwardRef>
                    Build timeline
                  </ForwardRef>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "Build timeline",
                    "src": "logo.png",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": "Build timeline",
                  },
                  "ref": null,
                  "rendered": "Build timeline",
                  "type": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-bZQynM",
                      "isStatic": true,
                      "rules": Array [
                        "
    margin-left: 10px;
    width: 100%;
",
                      ],
                    },
                    "displayName": "styled.span",
                    "render": [Function],
                    "styledComponentId": "sc-bZQynM",
                    "target": "span",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  },
                },
              ],
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-gzVnrw",
                  "isStatic": true,
                  "rules": Array [
                    "
    display: flex;
    align-items: center;
",
                  ],
                },
                "displayName": "styled.div",
                "render": [Function],
                "styledComponentId": "sc-gzVnrw",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              },
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <ForwardRef>
                  Back to Jenkins
                </ForwardRef>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": "Back to Jenkins",
                  "href": undefined,
                },
                "ref": null,
                "rendered": "Back to Jenkins",
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-htoDjs",
                    "isStatic": true,
                    "rules": Array [
                      "
    padding: 10px;
    border: 3px solid rgba(0,0,0,0.3);
    background-color: transparent;
    cursor: pointer;
    transition: 0.5s;
    color: ",
                      "#fff",
                      ";

    &:hover {
        background-color: rgba(0,0,0,0.3);
        border-color: transparent;
    }
",
                    ],
                  },
                  "displayName": "styled.button",
                  "render": [Function],
                  "styledComponentId": "sc-htoDjs",
                  "target": "button",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                },
              },
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bxivhb",
                  "isStatic": true,
                  "rules": Array [
                    "
",
                  ],
                },
                "displayName": "styled.span",
                "render": [Function],
                "styledComponentId": "sc-bxivhb",
                "target": "span",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              },
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-EHOje",
              "isStatic": true,
              "rules": Array [
                "
    box-shadow: inset 9999px 0 0 0 rgba(0, 0, 0, 0.2);
    padding: 10px;
    display: flex;
    align-items: center;
    justify-content: space-between;
",
              ],
            },
            "displayName": "styled.div",
            "render": [Function],
            "styledComponentId": "sc-EHOje",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <ForwardRef>
                Status: undefined
              </ForwardRef>,
              <ForwardRef>
                Started on undefined
              </ForwardRef>,
              <ForwardRef>
                Ran for 
              </ForwardRef>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Status: undefined",
              },
              "ref": null,
              "rendered": "Status: undefined",
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bxivhb",
                  "isStatic": true,
                  "rules": Array [
                    "
",
                  ],
                },
                "displayName": "styled.span",
                "render": [Function],
                "styledComponentId": "sc-bxivhb",
                "target": "span",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              },
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Started on undefined",
              },
              "ref": null,
              "rendered": "Started on undefined",
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bxivhb",
                  "isStatic": true,
                  "rules": Array [
                    "
",
                  ],
                },
                "displayName": "styled.span",
                "render": [Function],
                "styledComponentId": "sc-bxivhb",
                "target": "span",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              },
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Ran for ",
              },
              "ref": null,
              "rendered": "Ran for ",
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bxivhb",
                  "isStatic": true,
                  "rules": Array [
                    "
",
                  ],
                },
                "displayName": "styled.span",
                "render": [Function],
                "styledComponentId": "sc-bxivhb",
                "target": "span",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              },
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bwzfXH",
              "isStatic": true,
              "rules": Array [
                "
    display: flex;
    justify-content: space-between;
    padding: 10px;
",
              ],
            },
            "displayName": "styled.div",
            "render": [Function],
            "styledComponentId": "sc-bwzfXH",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <ForwardRef>
                Status: undefined
              </ForwardRef>,
              <ForwardRef>
                Started on undefined
              </ForwardRef>,
              <ForwardRef>
                Ran for 
              </ForwardRef>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Status: undefined",
              },
              "ref": null,
              "rendered": "Status: undefined",
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bxivhb",
                  "isStatic": true,
                  "rules": Array [
                    "
",
                  ],
                },
                "displayName": "styled.span",
                "render": [Function],
                "styledComponentId": "sc-bxivhb",
                "target": "span",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              },
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Started on undefined",
              },
              "ref": null,
              "rendered": "Started on undefined",
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bxivhb",
                  "isStatic": true,
                  "rules": Array [
                    "
",
                  ],
                },
                "displayName": "styled.span",
                "render": [Function],
                "styledComponentId": "sc-bxivhb",
                "target": "span",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              },
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Ran for ",
              },
              "ref": null,
              "rendered": "Ran for ",
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bxivhb",
                  "isStatic": true,
                  "rules": Array [
                    "
",
                  ],
                },
                "displayName": "styled.span",
                "render": [Function],
                "styledComponentId": "sc-bxivhb",
                "target": "span",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              },
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bwzfXH",
              "isStatic": true,
              "rules": Array [
                "
    display: flex;
    justify-content: space-between;
    padding: 10px;
",
              ],
            },
            "displayName": "styled.div",
            "render": [Function],
            "styledComponentId": "sc-bwzfXH",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": false,
          "rules": Array [
            "
    background-color: ",
            [Function],
            ";
    color: ",
            "#fff",
            ";

    font-family: 'Source Sans Pro', sans-serif;
    display: flex;
    flex-direction: column;
",
          ],
        },
        "displayName": "styled.header",
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "header",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
